services/checks/registry.go

```diff
@@ -66,6 +66,10 @@ func (r *registry) collect() ammodels.PostableAlerts {
 	return alerts
 }
 
+func (r *registry) getCheckResults() []sttCheckResult {
+	return r.checkResults
```

Comment:
We need locks there like in `collect()`

Reply:
That's not enough. `r.checksResults` is a slice (i.e. pointer), so you should return a copy of the slice. Otherwise, there will be race, anyone can get direct pointer to `r.checkResults` and change it's values without synchronization.

Take a look how it's made in `getMySQLChecks()`